test/kaniko_task_test.go

```diff
@@ -0,0 +1,185 @@
+	}
+}
+
+func getTaskRun(namespace string) *v1alpha1.TaskRun {
+	return &v1alpha1.TaskRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      kanikoTaskRunName,
+		},
+		Spec: v1alpha1.TaskRunSpec{
+			TaskRef: v1alpha1.TaskRef{
+				Name: kanikoTaskName,
+			},
+			Trigger: v1alpha1.TaskTrigger{
+				TriggerRef: v1alpha1.TaskTriggerRef{
+					Type: v1alpha1.TaskTriggerTypeManual,
+				},
+			},
+			Inputs: v1alpha1.TaskRunInputs{
+				Resources: []v1alpha1.PipelineResourceVersion{
+					v1alpha1.PipelineResourceVersion{
+						ResourceRef: v1alpha1.PipelineResourceRef{
+							Name: kanikoResourceName,
+						},
+						Version: "master",
+					},
+				},
+			},
+		},
+	}
+}
+
+// TestTaskRun is an integration test that will verify a TaskRun using kaniko
+func TestKanikoTaskRun(t *testing.T) {
+	logger := logging.GetContextLogger(t.Name())
+	c, namespace := setup(t, logger)
+
+	knativetest.CleanupOnInterrupt(func() { tearDown(logger, c.KubeClient, namespace) }, logger)
+	defer tearDown(logger, c.KubeClient, namespace)
+
+	if _, err := c.PipelineResourceClient.Create(getGitResource(namespace)); err != nil {
+		t.Fatalf("Failed to create Pipeline Resource `%s`: %s", kanikoResourceName, err)
+	}
+
+	// Create task
+	if _, err := c.TaskClient.Create(getTask(namespace, t)); err != nil {
+		t.Fatalf("Failed to create Task `%s`: %s", kanikoTaskName, err)
+	}
+
+	// Create TaskRun
+	if _, err := c.TaskRunClient.Create(getTaskRun(namespace)); err != nil {
+		t.Fatalf("Failed to create TaskRun `%s`: %s", kanikoTaskRunName, err)
+	}
+
+	// Verify status of TaskRun (wait for it)
+	if err := WaitForTaskRunState(c, kanikoTaskRunName, func(tr *v1alpha1.TaskRun) (bool, error) {
+		if len(tr.Status.Conditions) > 0 && tr.Status.Conditions[0].Status == corev1.ConditionTrue {
+			return true, nil
+		}
+		return false, nil
+	}, "TaskRunCompleted"); err != nil {
+		t.Errorf("Error waiting for TaskRun %s to finish: %s", kanikoTaskRunName, err)
+	}
+
+	// The Build created by the TaskRun will have the same name
+	b, err := c.BuildClient.Get(kanikoTaskRunName, metav1.GetOptions{})
+	if err != nil {
+		t.Errorf("Expected there to be a Build with the same name as TaskRun %s but got error: %s", kanikoTaskRunName, err)
+	}
+	cluster := b.Status.Cluster
+	if cluster == nil || cluster.PodName == "" {
+		t.Fatalf("Expected build status to have a podname but it didn't!")
+	}
+	podName := cluster.PodName
+	pods := c.KubeClient.Kube.CoreV1().Pods(namespace)
+	t.Logf("Retrieved pods for podname %s: %s\n", podName, pods)
+
+	req := pods.GetLogs(podName, &corev1.PodLogOptions{})
+	readCloser, err := req.Stream()
+	if err != nil {
+		t.Fatalf("Failed to open stream to read: %v", err)
+	}
+	defer readCloser.Close()
+	var buf bytes.Buffer
+	out := bufio.NewWriter(&buf)
+	_, err = io.Copy(out, readCloser)
+	if !strings.Contains(buf.String(), kanikoBuildOutput) {
+		t.Fatalf("Expected output %s from pod %s but got %s", kanikoBuildOutput, podName, buf.String())
+	}
```

Comment:
i feel like lines 161 - here should really be in a separate function (probably should have been from the beginning!)

Reply:
I feel its easier to follow the assertions in a test when it within the test

Reply:
i agree with keeping the assertions within the test, im more concerned about the logic to get the logs for the pod which i consider to be an implementation detail that isnt too important to the test itself and would be easier to follow if abstracted out

but anyway this log stuff in flux! in #133 @aaron-prindle is working around us not being able to get init logs by using volumes, so you could use that here if you wanted - however i started trying to use it in #128 and i think it needs a couple tweaks before it's reusable (almost there tho)

Reply:
ah i see, in #133 we are doing the assertions within that message. well i think we can get the best of both worlds, imma make a couple changes in #128

Reply:
okay after thinking about this more, I don't think every integration test needs to look for build output. I think that we should leave that for one test only to do (the existing test) and the rest of the tests can check functionality that's more specific to what they are testing.

like in this case, checking that the image was actually built - though since we can't actually push the image yet, i think we can wait for https://github.com/knative/build-pipeline/issues/144 to add that check

so what do you think about completely removing these lines instead?

Reply:
but what would the test be checking then? can we leave it until we have the better way then refactor it?

Reply:
> but what would the test be checking then?
It would pretty much assume that a successful status on the taskrun means it succeed (in fact, the current version isnt too much better - we assume that the build printing "build succeeded" is success but we don't actually know what it did :S)

> can we leave it until we have the better way then refactor it?
kk if you prefer, sound good!

Reply:
removed the no push, I think the original problem was in the creation of the image and not the upload